src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,187 @@
+
+////////////////////////////////////////////////////////////////////////////////
+//                                                                            //
+//    Methodology for Calculating Target Supply from Target Rate Per Block    //
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+// apy to rate per block off-chain conversion info:
+//         apy = ((1 + (borrowRatePerBlock / WAD) * blocksPerDay) ^ 365 - 1) * 100
+// (0)     borrowRatePerBlock = ((((apy / 100) + 1) ^ (1 / 365) - 1) / blocksPerDay) * WAD
+//         ATTOW the formula matches the borrow apy on (https://compound.finance/markets/DAI) using blocksPerDay = 6570
+//
+// https://github.com/compound-finance/compound-protocol/blob/master/contracts/BaseJumpRateModelV2.sol#L95
+//
+// util > kink:
+//         normalRate = kink * multiplierPerBlock / WAD + baseRatePerBlock;
+//         targetInterestRate = normalRate + jumpMultiplierPerBlock * (util - kink) / WAD
+//         util * jumpMultiplierPerBlock = (targetInterestRate - normalRate) * WAD + kink * jumpMultiplierPerBlock
+// (1)     util = kink + (targetInterestRate - normalRate) * WAD / jumpMultiplierPerBlock
+//
+// util <= kink:
+//         targetInterestRate = util * multiplierPerBlock / WAD + baseRatePerBlock
+// (2)     util = (targetInterestRate - baseRatePerBlock) * WAD / multiplierPerBlock
+//
+// https://github.com/compound-finance/compound-protocol/blob/master/contracts/BaseJumpRateModelV2.sol#L79
+//
+//         util = borrows * WAD / (cash + borrows - reserves);
+// (3)     cash + borrows - reserves = borrows * WAD / util
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPlan.sol";
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/CErc20.sol#L14
+interface CErc20Like {
+    function totalBorrows()           external view returns (uint256);
+    function totalReserves()          external view returns (uint256);
+    function interestRateModel()      external view returns (address);
+    function getCash()                external view returns (uint256);
+    function underlying()             external view returns (address);
+}
+
+// https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/BaseJumpRateModelV2.sol#L10
+interface InterestRateModelLike {
+    function baseRatePerBlock()       external view returns (uint256);
+    function kink()                   external view returns (uint256);
+    function multiplierPerBlock()     external view returns (uint256);
+    function jumpMultiplierPerBlock() external view returns (uint256);
+}
+
+contract D3MCompoundPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    InterestRateModelLike        public tack;
+    uint256                      public barb; // target Interest Rate Per Block [wad] (0)
```

Comment:
What does this number look like? If governance wants to target, say, a 2.75% annual rate, what's the formula for calculating that? We might need to document this formula.

Reply:
Ideally it would be nice to have an `external pure` function in this Plan such that a developer can pass basis points as an argument and get an appropriate value to be used for setting `barb`

Reply:
This is not a trivial calculation and adding it on-chain should be discussed separately.
If we do add it a proper place can also be dss-exec-lib.
This is not a blocker as we can start with a manual calculation or a rates translation chart.